drivers/bluetooth/hci/hci_da1469x.c

```diff
@@ -0,0 +1,495 @@
+	net_buf_put(&rx.fifo, buf);
+}
+
+static inline void read_header(void)
+{
+	switch (rx.type) {
+	case H4_NONE:
+		h4_get_type();
+		return;
+	case H4_EVT:
+		get_evt_hdr();
+		break;
+	case H4_ACL:
+		get_acl_hdr();
+		break;
+	case H4_ISO:
+		if (IS_ENABLED(CONFIG_BT_ISO)) {
+			get_iso_hdr();
+			break;
+		}
+		__fallthrough;
+	default:
+		CODE_UNREACHABLE;
+		return;
+	}
+
+	if (rx.have_hdr && rx.buf) {
+		if (rx.remaining > net_buf_tailroom(rx.buf)) {
+			LOG_ERR("Not enough space in buffer");
+			rx.discard = rx.remaining;
+			reset_rx();
+		} else {
+			copy_hdr(rx.buf);
+		}
+	}
+}
+
+static inline void process_rx(void)
+{
+	LOG_DBG("remaining %u discard %u have_hdr %u rx.buf %p len %u", rx.remaining, rx.discard,
+		rx.have_hdr, rx.buf, rx.buf ? rx.buf->len : 0);
+
+	if (rx.discard) {
+		rx.discard -= h4_discard(rx.discard);
+		return;
+	}
+
+	if (rx.have_hdr) {
+		read_payload();
+	} else {
+		read_header();
+	}
+}
+
+void cmac_read_req(void)
+{
+	while (cmac_mbox_has_data()) {
+		process_rx();
+	}
+}
+
+void cmac_rng_req(void)
+{
+	k_sem_give(&rng_sem);
+}
+
+static void rng_thread(void *p1, void *p2, void *p3)
+{
+	uint32_t word;
+
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	while (1) {
+		k_sem_take(&rng_sem, K_FOREVER);
+
+		while (cmac_rand_needs_data()) {
+			word = sys_rand32_get();
+			cmac_rand_fill(&word, 1);
+		}
+
+		cmac_signal();
+	}
+}
+
+static int bt_da1469x_open(void)
+{
+	k_tid_t tid;
+
+	tid = k_thread_create(&rx_thread_data, rx_thread_stack,
+			      K_KERNEL_STACK_SIZEOF(rx_thread_stack),
+			      rx_thread, NULL, NULL, NULL,
+			      K_PRIO_COOP(CONFIG_BT_RX_PRIO),
+			      0, K_NO_WAIT);
+	k_thread_name_set(tid, "bt_rx_thread");
+
+	k_sem_init(&rng_sem, 0, 1);
+
+	tid = k_thread_create(&rng_thread_data, rng_thread_stack,
+			      K_KERNEL_STACK_SIZEOF(rng_thread_stack),
+			      rng_thread, NULL, NULL, NULL,
+			      K_PRIO_COOP(CONFIG_BT_RX_PRIO),
+			      0, K_NO_WAIT);
+	k_thread_name_set(tid, "bt_rng_thread");
+
+	cmac_enable();
+	irq_enable(CMAC2SYS_IRQn);
+
+	return 0;
+}
+
+#ifdef CONFIG_BT_HCI_HOST
+static int bt_da1469x_close(void)
+{
+	irq_disable(CMAC2SYS_IRQn);
+	cmac_disable();
+
+	return 0;
+}
+#endif /* CONFIG_BT_HCI_HOST */
+
+static int bt_da1469x_send(struct net_buf *buf)
+{
+	switch (bt_buf_get_type(buf)) {
+	case BT_BUF_ACL_OUT:
+		LOG_DBG("ACL: buf %p type %u len %u", buf, bt_buf_get_type(buf), buf->len);
+		net_buf_push_u8(buf, H4_ACL);
+		break;
+	case BT_BUF_CMD:
+		LOG_DBG("CMD: buf %p type %u len %u", buf, bt_buf_get_type(buf), buf->len);
+		net_buf_push_u8(buf, H4_CMD);
+		break;
+	default:
+		LOG_ERR("Unsupported type");
+		return -EINVAL;
+	}
+
+	cmac_mbox_write(buf->data, buf->len);
+
+	net_buf_unref(buf);
+
+	return 0;
+}
+
+static const struct bt_hci_driver drv = {
+	.name           = "BT DA1469x",
+	.bus            = BT_HCI_DRIVER_BUS_IPM,
+	.open           = bt_da1469x_open,
+#ifdef CONFIG_BT_HCI_HOST
+	.close          = bt_da1469x_close,
+#endif
```

Comment:
Not directly related to this PR, but it seems a bit arbitrary that `close()` would depend on the Zephyr host stack. The only other user of the HCI driver API right now. is HCI_RAW, which indeed doesn't use `close()`, but it seems wrong to me that the HCI driver API makes such assumptions about the higher level implementation that uses the API. I guess we can fix this once we move to a native Zephyr driver API.

Reply:
@andrzej-kaczmarek could I get some comment here? Drivers shouldn't generally know or care who their user is (the normal bluetooth host stack or HCI_RAW). So I think it might be cleaner to just remove the #ifdefs from close().